docs/changelog/84263.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 84263
+summary: Capture client's effective realm for token creation
```

Comment:
I don't think this description makes sense for release notes

Reply:
I updated it to 
> Ensure tokens represent effective user's identity in all cases
Let me know if this works for you.